modules/pricer/src/test/java/com/opengamma/strata/pricer/fx/BlackFxSingleBarrierOptionProductPricerTest.java

```diff
@@ -0,0 +1,678 @@
+    assertEquals(computedPut.getSensitivity(), -expectedPut * NOTIONAL, TOL * NOTIONAL);
+    assertEquals(computedPut.getCurrency(), USD);
+    assertEquals(computedPut.getCurrencyPair(), CURRENCY_PAIR);
+    assertEquals(computedPut.getStrike(), STRIKE_RATE);
+    assertEquals(computedPut.getForward(), forward, TOL);
+    assertEquals(computedPut.getExpiryDateTime(), EXPIRY_DATETIME);
+  }
+
+  public void test_vega_presentValueSensitivityVolatility_atExpiry() {
+    double computedVegaCall = PRICER.vega(CALL_UKI, RATE_PROVIDER_EXPIRY, VOL_PROVIDER_EXPIRY);
+    PointSensitivityBuilder computedCall =
+        PRICER.presentValueSensitivityVolatility(CALL_UKI, RATE_PROVIDER_EXPIRY, VOL_PROVIDER_EXPIRY);
+    double computedVegaPut = PRICER.vega(PUT_UKO_BASE, RATE_PROVIDER_EXPIRY, VOL_PROVIDER_EXPIRY);
+    PointSensitivityBuilder computedPut =
+        PRICER.presentValueSensitivityVolatility(PUT_UKO_BASE, RATE_PROVIDER_EXPIRY, VOL_PROVIDER_EXPIRY);
+    assertEquals(computedVegaCall, 0d);
+    assertEquals(computedCall, PointSensitivityBuilder.none());
+    assertEquals(computedVegaPut, 0d);
+    assertEquals(computedPut, PointSensitivityBuilder.none());
+  }
+
+  public void test_vega_presentValueSensitivityVolatility_afterExpiry() {
+    double computedVegaCall = PRICER.vega(CALL_UKI, RATE_PROVIDER_AFTER, VOL_PROVIDER_AFTER);
+    PointSensitivityBuilder computedCall =
+        PRICER.presentValueSensitivityVolatility(CALL_UKI, RATE_PROVIDER_AFTER, VOL_PROVIDER_AFTER);
+    double computedVegaPut = PRICER.vega(PUT_UKO_BASE, RATE_PROVIDER_AFTER, VOL_PROVIDER_AFTER);
+    PointSensitivityBuilder computedPut =
+        PRICER.presentValueSensitivityVolatility(PUT_UKO_BASE, RATE_PROVIDER_AFTER, VOL_PROVIDER_AFTER);
+    assertEquals(computedVegaCall, 0d);
+    assertEquals(computedCall, PointSensitivityBuilder.none());
+    assertEquals(computedVegaPut, 0d);
+    assertEquals(computedPut, PointSensitivityBuilder.none());
+  }
+
+  //-------------------------------------------------------------------------
+  public void test_currencyExposure() {
+    for (ResolvedFxSingleBarrierOption option : OPTION_ALL) {
+      CurrencyAmount pv = PRICER.presentValue(option, RATE_PROVIDER, VOL_PROVIDER);
+      MultiCurrencyAmount computed = PRICER.currencyExposure(option, RATE_PROVIDER, VOL_PROVIDER);
+      FxMatrix fxMatrix = FxMatrix.builder().addRate(EUR, USD, SPOT + FD_EPS).build();
+      ImmutableRatesProvider provBumped = RATE_PROVIDER.toBuilder().fxRateProvider(fxMatrix).build();
+      CurrencyAmount pvBumped = PRICER.presentValue(option, provBumped, VOL_PROVIDER);
+      double ceCounterFD = pvBumped.getAmount() - pv.getAmount();
+      double ceBaseFD = pvBumped.getAmount() * (SPOT + FD_EPS) - pv.getAmount() * SPOT;
+      assertEquals(computed.getAmount(EUR).getAmount() * FD_EPS, ceCounterFD, NOTIONAL * FD_EPS);
+      assertEquals(-(computed.getAmount(USD).getAmount() - pv.getAmount()) * FD_EPS, ceBaseFD, NOTIONAL * FD_EPS);
+    }
```

Comment:
The last 3 lines of the test are incorrect. Replace by:

      double ceBaseFD = pvBumped.getAmount() / (SPOT + FD_EPS) - pv.getAmount() / SPOT;
      assertEquals(computed.getAmount(EUR).getAmount() * FD_EPS, ceCounterFD, NOTIONAL * TOL);
      assertEquals(computed.getAmount(USD).getAmount() * (1.0d/ (SPOT + FD_EPS) - 1.0d/ SPOT), ceBaseFD, NOTIONAL * TOL);